cpp/include/cudf/copying.hpp

```diff
@@ -173,5 +174,31 @@ std::vector<column_view> slice(column_view const& input,
 std::vector<column_view> split(column_view const& input,
                                std::vector<size_type> const& splits);
 
+/**
+* @brief Creates a new `table` similar to the input but with values in every
+* column offset by `periods`, using `fill_value` when input is out-of-bounds.
+*
+* When `i + periods` is in bounds:
+*   `out[col][i] = in[col][i + periods]`
+*
+* When `i + periods` is out of bounds and `fill_value` is supplied:
+*   `out[col][i] = fill_value[i]`
+*
+* When `i + periods` is out of bounds and `fill_value is not supplied:
+*   `out[col][i] = null`
```

Comment:
Examples are much better at explaining what's happening.

Reply:
Addressing these concerns in #3662.